src/components/tx-flow/SafeTxProvider.tsx

```diff
@@ -36,15 +37,18 @@ const SafeTxProvider = ({ children }: { children: ReactNode }): ReactElement =>
   const [nonceNeeded, setNonceNeeded] = useState<boolean>(true)
   const [safeTxGas, setSafeTxGas] = useState<number>()
 
+  const { safe } = useSafeInfo()
+
   // Signed txs cannot be updated
   const isSigned = safeTx && safeTx.signatures.size > 0
 
   // Recommended nonce and safeTxGas
   const recommendedNonce = useRecommendedNonce()
   const recommendedSafeTxGas = useSafeTxGas(safeTx)
 
-  // Priority to external nonce, then to the recommended one
-  const finalNonce = isSigned ? safeTx?.data.nonce : nonce ?? recommendedNonce ?? safeTx?.data.nonce
+  // Priority to external nonce if valid, then to the recommended one
+  const isNonceValid = nonce !== undefined && nonce >= safe.nonce
+  const finalNonce = isSigned ? safeTx?.data.nonce : isNonceValid ? nonce : recommendedNonce ?? safeTx?.data.nonce
```

Comment:
`nonce` is the manually set nonce. It will likely not be there in the use case you're trying to fix.
You should be checking `recommendedNonce` for validity instead.

I would do something like

```js
const finalNonce = isSigned ? safeTx?.data.nonce : nonce : recommendedNonce >= safe.nonce ? recommendedNonce : safe.nonce
```

Reply:
I think `nonce` is automatically set in the beginning to what `recommendedNonce` is. I think its a bit confusing to check against `safe.nonce` but then return `recommendedNonce` or vice versa. From my tests, `safe.nonce` updates a bit sooner than `recommendedNonce` but I am not sure why. We should try to understand how these values connect and only use one of them.

Reply:
> I think nonce is automatically set in the beginning to what recommendedNonce is

It's not.

Reply:
Sorry, I should've clarified. The Provider returns the `recommendedNonce` as the `finalNonce` and the form uses that as the `defaultValue`. This runs the form validation where we call `setNonce` which then updates `nonce` to what the  `recommendedNonce` is at the time.

Reply:
I see. That's bad. `setNonce` should only be called if the user deliberately edited the nonce.

That's why the solution with `setValue` in this PR is no bueno.

Reply:
Not changing the form's field value but only triggering validation leads to weird states:
- Indeed no wrong / too low nonce will be used in the `SafeTxProvider` `finalNonce` (which is good)
- But: Our UI still shows that you use a wrong nonce.

Here is a screenshot showing a nonce error while using the actual updated `recommendedNonce` in the `SafeTxProvider` and thus enabling the sign button.
When signing a transaction the updated higher nonce is being used. 
![Screenshot 2023-08-14 at 16 07 21](https://github.com/safe-global/safe-wallet-web/assets/2670790/f2a0fded-e284-4508-affb-8cdeaa350116)

I think we form's nonce value should always be in sync with the `finalNonce` from the `SafeTxProvider` which will be used for the transaction.

I now changed the behavior slightly:
- Don't trigger form validation and `setNonce` for the `defaultValues` (so `nonce` is undefined in the SafeTxProvider when computing the `finalNonce`)
- When the `finalNonce` updates, we propagate the change to the TxNonce field through the `values` property of `useForm`:
```ts
  const formMethods = useForm({
    defaultValues: {
      [TxNonceFormFieldNames.NONCE]: nonce,
    },
    mode: 'onChange',
    values: {
      [TxNonceFormFieldNames.NONCE]: nonce,
    },
  })
```

Like this the validation and `finalNonce` computations happens in `SafeTxProvider` and we will never store an invalid nonce as finalNonce. Plus the valid `finalNonce` will always be displayed in the UI if it is updated.

Reply:
The form code looks good now but you still need to fix the provider code. I would restore it to how it was and move the safe.nonce thing to useRecommendedNonce.